### PR TITLE
send things to codecov.io

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

File: run-checks [hunk 1]
```diff
@@ -140,10 +140,12 @@ if [ "$UNIT" = 1 ]; then
     echo Running tests from $(pwd)
     for pkg in $(go list ./... | grep -v '/vendor/' ); do
         go test -i $pkg
-        $goctest -v -coverprofile=.coverage/profile.out $pkg
+        $goctest -v -coverprofile=.coverage/profile.out -covermode=atomic $pkg
         append_coverage .coverage/profile.out
     done
 
+    # copy in place for codecov.io
+    cp .coverae/profile.out coverage.txt
```
Comment (zyga) [inline run-checks]: This fails with: `cp: cannot stat '.coverae/profile.out': No such file or directory`

**Comment by chipaca** [comment]
So, one question I have for this is how does the coverage file make its way back from spread to travis.

File: .travis.yml [hunk 2]
```diff
@@ -19,4 +19,5 @@ install:
 
 script:
     - ./run-checks --static || travis_terminate 1
+    - ./run-checks --unit || travis_terminate 1
```
Comment (fgimenez) [inline .travis.yml]: If we readd unit tests here, shouldn't we remove the uni tests task?

Comment (fgimenez) [inline .travis.yml]: Sorry, this is the task https://github.com/snapcore/snapd/blob/master/tests/unit/go/task.yaml

File: run-checks [hunk 3]
```diff
@@ -140,10 +142,14 @@ if [ "$UNIT" = 1 ]; then
...
     for pkg in $(go list ./... | grep -v '/vendor/' ); do
         go test -i $pkg
-        $goctest -v -coverprofile=.coverage/profile.out $pkg
+        $goctest -v -coverprofile=.coverage/profile.out -covermode=atomic $pkg
         append_coverage .coverage/profile.out
     done
 
+    # upload to codecov.io if on travis
+    if [ "${TRAVIS_BUILD_NUMBER:-}" ]; then
+        bash <(curl -s https://codecov.io/bash) -f .coverage/coverage.out
```
Comment (pedronis) [inline run-checks]: woa! (I know we don't have commit bits inside travis so should be ok but looks a bit scary anyway)

Comment (mvo5) [inline run-checks]: We can also move it into .travis.yml into after_success. this way we can be sure it is not run by accident. And yes, it looks scary.

File: .travis.yml (hunk 2 above)
Comment (mvo5) [inline .travis.yml]: I think thats a good idea. The downside of the unittest as an extra step is that it increases the test time a little bit. But the upside is that failing unittests will be noticed right away.

File: run-checks (hunk 3 above)
Comment (zyga) [inline run-checks]: We should talk to them about this thing that's pretty cool nowadays. I think it's called _snaps_

**Pull request merged**